ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/query/VaultNamedQueryBuilderFactory.java

```diff
@@ -0,0 +1,62 @@
+package net.corda.v5.ledger.common.query;
+
+import net.corda.v5.base.annotations.DoNotImplement;
+import net.corda.v5.base.annotations.Suspendable;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A builder that is used to create and build a {@link VaultNamedQuery}.
+ * <p>
+ * The register method is called when the build is finished so that the query is stored and can be fetched and executed
+ * later on.
+ */
+@DoNotImplement
+public interface VaultNamedQueryBuilderFactory {
+
+    /**
+     * Sets the name of the named query
+     * @param queryName Name of the query
+     * @return A builder instance with the name set
+     */
+    @Suspendable
+    @NotNull VaultNamedQueryBuilderFactory create(@NotNull String queryName);
+
+    /**
+     * Sets the where clause of the named query
+     * @param json The json query representation
+     * @return A builder instance with the where clause set
+     */
+    @Suspendable
+    @NotNull VaultNamedQueryBuilderFactory whereJson(@NotNull String json);
```

Comment:
Can we have a separate `VaultNamedQueryBuilder` that `create` returns. Then the rest of the functions exist on that interface instead.

Reply:
I'm happy to make that change but do you think it's worth having one more interface just for a `create` function? I mean I don't have a strong opinion but not sure if it's worth it.